tests/test_token_budget.py

```
@@ -0,0 +1,118 @@
+        cfg = load_budget_config(str(_REAL_YAML))
+        base = compute_max_tokens("code-edit", 200, 131072,
+                                  tier=4,
+                                  tier_multiplier=cfg["tier_multiplier"])
+        self.assertEqual(base, 2048)
+        tier0 = compute_max_tokens("code-edit", 200, 131072,
+                                   tier=0,
+                                   tier_multiplier=cfg["tier_multiplier"])
+        self.assertEqual(tier0, 1024)
+
+    def test_env_override(self):
+        os.environ["TOKEN_BUDGET_OVERRIDES"] = '{"reasoning": 1234}'
```

Ruff (0.16.1)

[error] Possible hardcoded password assigned to: "TOKEN_BUDGET_OVERRIDES"

(S105)